accumulo/src/main/java/org/vertexium/accumulo/AccumuloElement.java

```diff
@@ -250,16 +243,9 @@ public Property getProperty(String name) {
 
     @Override
     public Object getPropertyValue(String name, int index) {
-        if (ID_PROPERTY_NAME.equals(name)) {
-            return getIdProperty();
-        } else if (Edge.LABEL_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            return getEdgeLabelProperty();
-        } else if (Edge.OUT_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            return getOutVertexIdProperty();
-        } else if (Edge.IN_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            return getInVertexIdProperty();
-        } else if (Edge.IN_OR_OUT_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            return getInOrOutVertexIdProperty();
+        Property reservedProperty = getReservedProperty(name);
+        if (reservedProperty != null) {
+            return reservedProperty.getValue();
         }
         getFetchHints().assertPropertyIncluded(name);
```

Comment:
Move this check into `getPropertyValue(String key, String name, int index)`, gut this method and have it just call the other one.